extensions/amp-story/1.0/bookend/components/article.js

```diff
@@ -121,7 +126,16 @@ export class ArticleComponent {
           </div>`;
 
       const {image} = htmlRefs(imgEl);
-      addAttributesToElement(image, dict({'src': articleData.image}));
+
+      if (isServedFromCache(doc) && !isAbsoluteUrl(articleData.image)) {
+        const fullUrl = doc.location.href + articleData.image;
+        // TODO(Enriqe): add extra params for resized image, for example:
+        // (/ii/w${width}/s)
+        const optimizedUrl = fullUrl.replace('/c/s/', '/i/s/');
+        addAttributesToElement(image, dict({'src': optimizedUrl}));
+      } else {
+        addAttributesToElement(image, dict({'src': articleData.image}));
```

Comment:
Please use a reassignable variable to store the URL so you can keep only one call to `addAttributesToElement()`, to minimize code duplication